python/vowpalwabbit/pyvw.py

```diff
@@ -241,10 +241,19 @@ def format_input(key, val):
 
         pylibvw.vw.__init__(self, " ".join(l))
 
+        self.parser_ran = False
+
         # check to see if native parser needs to run
         ext_file_args = ["d", "data", "passes"]
         if any(x in kw for x in ext_file_args):
             pylibvw.vw.run_parser(self)
+            self.parser_ran = True
+        elif arg_str:
+            # space after -d to avoid matching with other substrings
+            ext_file_cmd_str = ["-d ", "--data", "--passes"]
+            if [cmd for cmd in ext_file_cmd_str if(cmd in arg_str)]:
```

Comment:
should this be `any([cmd for cmd in ext_file_cmd_str if(cmd in arg_str)]` ?

Reply:
Doesn't need to, the expression after `if` will create an array that contains the args that are present, so if any of them are present the array will not be empty, so we will enter the `if` statement once